Fixed bug in broadcaster where an actor would be added to two buckets

bogdan-dumitrescu opened the pull request:
Also added rehashing logic.

jwijgerd commented on broadcast/src/main/java/org/elasticsoftware/elasticactors/broadcast/handlers/RehashHandlers.java:
```diff
@@ -0,0 +1,110 @@
+    }
+
+    @Message(serializationFramework = JacksonSerializationFramework.class, durable = true, immutable = true)
+    private static final class RehashRequestInternal {
```
Hi Bogdan, I think it would be best to have this class at the highest level as I am not 100% sure that the MessagesScanner will correctly identity it

bogdan-dumitrescu commented:
Hi Joost,


I implemented your suggestion and also added caching & replaying of the messages received during a rehash. 


Best regards,

Bogdan,
